pkg/service/service.go

```diff
@@ -37,24 +37,6 @@ var (
 	addMetric    = metrics.ServicesCount.WithLabelValues("add")
 )
 
-// LBMap is the interface describing methods for manipulating service maps.
-type LBMap interface {
-	UpsertService(*lbmap.UpsertServiceParams) error
-	UpsertMaglevLookupTable(uint16, map[string]lb.BackendID, bool) error
-	IsMaglevLookupTableRecreated(bool) bool
-	DeleteService(lb.L3n4AddrID, int, bool, lb.SVCNatPolicy) error
-	AddBackend(lb.Backend, bool) error
-	UpdateBackendWithState(lb.Backend) error
-	DeleteBackendByID(lb.BackendID) error
-	AddAffinityMatch(uint16, lb.BackendID) error
-	DeleteAffinityMatch(uint16, lb.BackendID) error
-	UpdateSourceRanges(uint16, []*cidr.CIDR, []*cidr.CIDR, bool) error
-	DumpServiceMaps() ([]*lb.SVC, []error)
-	DumpBackendMaps() ([]*lb.Backend, error)
-	DumpAffinityMatches() (lbmap.BackendIDByServiceIDSet, error)
-	DumpSourceRanges(bool) (lbmap.SourceRangeSetByServiceID, error)
-}
-
```

Comment:
[1] The idea was that `pkg/service` wouldn't need to import the `lbmap` package. But I guess that cat is out of the box so you can disregard my comment of interfaces living in a package outside of their implementation.

Reply:
I wanted to move the LBMap ownership to datapath, so I needed the interface out from `pkg/service`. Initially I moved the LBMap interface to `pkg/datapath/types` but that felt silly as we had `pkg/maps/lbmap`. But I see your point. Perhaps this should be in `pkg/datapath/types` instead and then we'd have the concrete implementations in `pkg/maps/lbmap` and `pkg/testutils/mockmaps`. (longer term I'd like these to move under `pkg/datapath` as they're implementation details of the datapath).

Would `pkg/datapath/types` seem like a reasonable place for `LBMap` as it's logically part of the `Datapath` interface?

Reply:
> Would `pkg/datapath/types` seem like a reasonable place for `LBMap` as it's logically part of the `Datapath` interface?

@joamaki yes, I think that's perfect.

Reply:
Done. I also of course had to move `UpsertServiceParams` and few others there.